I'm seeing some regressions when I try to build selene with this PR for x86-windows-dbg. Can you take a look?

Here's the log of what I'm seeing:

```
[1/28] C:\VS2017\Stable\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x86\cl.exe   /TP -DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -Dselene_base_EXPORTS -IC:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648 -I. /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    /MP /permissive- /wd4324 /wd4611 /W4 -std:c++17 /showIncludes /Foselene\CMakeFiles\selene_base.dir\base\MessageLog.cpp.obj /Fdselene\CMakeFiles\selene_base.dir\ /FS -c C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp
FAILED: selene/CMakeFiles/selene_base.dir/base/MessageLog.cpp.obj 
C:\VS2017\Stable\VC\Tools\MSVC\14.16.27023\bin\Hostx64\x86\cl.exe   /TP -DBOOST_CONFIG_SUPPRESS_OUTDATED_MESSAGE -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS -Dselene_base_EXPORTS -IC:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648 -I. /DWIN32 /D_WINDOWS /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    /MP /permissive- /wd4324 /wd4611 /W4 -std:c++17 /showIncludes /Foselene\CMakeFiles\selene_base.dir\base\MessageLog.cpp.obj /Fdselene\CMakeFiles\selene_base.dir\ /FS -c C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.16.27023.1 for x86
Copyright (C) Microsoft Corporation.  All rights reserved.

C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(15): error C2375: 'sln::MessageLog::contains_warnings': redefinition; different linkage
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene/base/MessageLog.hpp(54): note: see declaration of 'sln::MessageLog::contains_warnings'
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(22): error C2375: 'sln::MessageLog::contains_errors': redefinition; different linkage
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene/base/MessageLog.hpp(55): note: see declaration of 'sln::MessageLog::contains_errors'
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(29): error C2375: 'sln::MessageLog::contains_warnings_or_errors': redefinition; different linkage
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene/base/MessageLog.hpp(56): note: see declaration of 'sln::MessageLog::contains_warnings_or_errors'
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(37): error C2375: 'sln::message_type_to_string': redefinition; different linkage
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene/base/MessageLog.hpp(64): note: see declaration of 'sln::message_type_to_string'
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(52): error C2375: 'sln::operator <<': redefinition; different linkage
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene/base/MessageLog.hpp(66): note: see declaration of 'sln::operator <<'
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(56): error C2065: 'message_type_to_string': undeclared identifier
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(56): error C2568: '<<': unable to resolve function overload
C:\vsts\_work\1\s\buildtrees\selene\src\v0.3-9d019b2648\selene\base\MessageLog.cpp(56): note: could be 'std::string sln::message_type_to_string(sln::MessageType)'
```

--- a/ports/selene/CONTROL
+++ b/ports/selene/CONTROL
@@ -1,4 +1,4 @@
 Source: selene
-Version: 0.3
+Version: 0.3.1
 Description: A C++17 image representation, processing and I/O library.
 Build-Depends: zlib, libpng, libjpeg-turbo, tiff

--- a/ports/selene/portfile.cmake
+++ b/ports/selene/portfile.cmake
@@ -1,10 +1,12 @@
 include(vcpkg_common_functions)
 
+vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO kmhofmann/selene
-    REF v0.3
-    SHA512 53a4a6577b4e618c5b080e0ddaa1e0e28b7c0a27e800eb9d1b1a2a6fbfaf630a6f326a3d070ad0c3f9bd8fb5e6bdd7fafbbd5a49e5f9a6f9ae79e0b50d20f741
+    REF v0.3.1
+    SHA512 7bc57ebe9e2442da2716eb5c1af11f8d610b0b09fe96e3122d1028575732b6045a987c499bbf7de53003edd627b8809d86c80ea4975fc2264a1c61d5891a46c3
     HEAD_REF master
 )
 

